driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java

```diff
@@ -262,28 +263,42 @@ protected Object convertValue(final OIndex<?> idx, Object iValue) {
     }
 
     public Stream<OrientVertex> getIndexedVertices(OIndex<Object> index, Optional<Object> valueOption) {
+        return getIndexedElements(index, valueOption, OrientVertex::new, OrientVertex::new);
+    }
+
+    public Stream<OrientEdge> getIndexedEdges(OIndex<Object> index, Optional<Object> valueOption) {
+        return getIndexedElements(index, valueOption, OrientEdge::new, OrientEdge::new);
+    }
+
+    private <ElementType extends OrientElement> Stream<ElementType> getIndexedElements(OIndex<Object> index, Optional<Object> valueOption,
+            BiFunction<OrientGraph, OIdentifiable, ElementType> newElementById, BiFunction<OrientGraph, ODocument, ElementType> newElementByDoc) {
```

Comment:
Since OrientEdge constructor checks if rawElement is instanceof ODocument, you could use only the OIdentifiable constructor, could you not?